pkg/testselect/testselect.go

```diff
@@ -0,0 +1,155 @@
+package testselect
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"log"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/openshift-knative/hack/pkg/prowgen"
+	"gopkg.in/yaml.v2"
+	"k8s.io/test-infra/prow/clonerefs"
+)
+
+const (
+	all = "All"
+)
+
+// TestSuites holds mapping between file path regular expressions and
+// test suites that cover the paths.
+type TestSuites struct {
+	List []TestSuite `yaml:"testsuites"`
+}
+
+type TestSuite struct {
+	Name 		 string   `yaml:"name"`
+	RunIfChanged []string `yaml:"run_if_changed"`
+	Tests 		 []string   `yaml:"tests"`
```

Comment:
Are those strings bash functions or makefile targets?

Reply:
nit: I think you need to reformat go code

Reply:
These are just "strings". The PR in serverless-operator uses names of bash functions so that it can be used like this:
```
function serverless_operator_kafka_e2e_tests {
  # Note: ${FUNCNAME[0]} resolves to "serverless_operator_kafka_e2e_tests"
  should_run "${FUNCNAME[0]}" || return  
  ...
}
```
It's basically a mapping between code path regular expressions and strings.

I will reformat the code

Reply:
thansk! maybe we can add your comment above on top of the `tests` field ?